### ISSUE TITLE
YAML front matter parser does not strip single or double quotes on scalars

### ISSUE THREAD

**Issue by JakeWharton** [commonmark/commonmark-java#260]
Steps to reproduce the problem (provide example Markdown if applicable):

```
---
title: 'Sixteen corners'
layout: post
---

Last year
```

Expected behavior:

```kotlin
frontMatter["title"].single() == "Sixteen corners"
```

Actual behavior:

<img width="311" alt="Screen Shot 2022-05-18 at 10 56 24 PM" src="https://user-images.githubusercontent.com/66577/169194712-3c3d1b5a-f89b-4326-8949-958ce379f555.png">

Note the retained single quotes (`'`) surrounding the value. This is also a problem with double quotes (`"`).

YAML spec dictates behavior on unquoted, single-quoted, and double-quoted scalars: https://yaml.org/spec/1.2.2/#73-flow-scalar-styles

**Comment by robinst** [comment]
Yeah you're right, that's broken. The reason for that is that it currently does some manual (and very limited) YAML parsing, see [`YamlFrontMatterBlockParser`](https://github.com/commonmark/commonmark-java/blob/main/commonmark-ext-yaml-front-matter/src/main/java/org/commonmark/ext/front/matter/internal/YamlFrontMatterBlockParser.java). We should probably never have done that, and just depend on a real YAML parser in that extension.

So two ways to fix this:

1. Extend our manual parser to handle quoting
2. Depend on a YAML library (which one?) instead, to be able to parse YAML 1.1 or 1.2 (which one?)

Do you have any opinions on those @JakeWharton?

**Comment by robinst** [comment]
Created a PR for 1., see #261.

**Comment by robinst** [comment]
Ok, so sounds like my PR will be enough to solve the issue for you?: https://github.com/commonmark/commonmark-java/pull/261/files#diff-3bf17f3edca5ff1728b4e1918880038dcbe66f231114c2a1e40729793d07bb96R286

**Comment by JakeWharton** [comment]
Yep! Looks good.

**Comment by robinst** [comment]
Alright, released this fix as 0.19.0 🎉: https://github.com/commonmark/commonmark-java/blob/main/CHANGELOG.md#0190---2022-06-02
